### PR TITLE
fix: tvOS support

### PR DESCRIPTION
**Pull request by Krizzu**
Summary:
---------

Bringing back tvOS support that was lost some time ago. Fixes https://github.com/react-native-community/async-storage/issues/416

Test Plan:
----------

Green CI.
Test AsyncStorage on tvOS

### CODE REVIEW

**Comment by Krizzu** [comment]
@tido64 Hey, mind having a quick look at this?

File: ios/RNCAsyncStorage.m [hunk 1]
```diff
@@ -94,10 +94,19 @@ static void RCTAppendError(NSDictionary *error, NSMutableArray<NSDictionary *> *
...
+  #if TARGET_OS_TV
+    storageDirectoryPath = NSSearchPathForDirectoriesInDomains(NSCachesDirectory, NSUserDomainMask, YES).firstObject;
+    storageDirectoryPath = [storageDirectoryPath stringByAppendingPathComponent:storageDir];
+  #else
+    storageDirectoryPath = NSSearchPathForDirectoriesInDomains(NSApplicationSupportDirectory, NSUserDomainMask, YES).firstObject;
+    // We should use the "Application Support/[bundleID]" folder for persistent data storage that's hidden from users
+    storageDirectoryPath = [storageDirectoryPath stringByAppendingPathComponent:[[NSBundle mainBundle] bundleIdentifier]]; // Per Apple's docs, all app content in Application Support must be within a subdirectory of the app's bundle identifier
+    storageDirectoryPath = [storageDirectoryPath stringByAppendingPathComponent:storageDir];
+  #endif
+
```
Comment (tido64) [inline ios/RNCAsyncStorage.m:108]: I think we can share the last line as well.
```suggestion
  #if TARGET_OS_TV
    storageDirectoryPath = NSSearchPathForDirectoriesInDomains(NSCachesDirectory, NSUserDomainMask, YES).firstObject;
  #else
    storageDirectoryPath = NSSearchPathForDirectoriesInDomains(NSApplicationSupportDirectory, NSUserDomainMask, YES).firstObject;
    // We should use the "Application Support/[bundleID]" folder for persistent data storage that's hidden from users
    storageDirectoryPath = [storageDirectoryPath stringByAppendingPathComponent:[[NSBundle mainBundle] bundleIdentifier]];
  #endif
    // Per Apple's docs, all app content in Application Support must be within a subdirectory of the app's bundle identifier
    storageDirectoryPath = [storageDirectoryPath stringByAppendingPathComponent:storageDir];
```

**Review by tido64 — APPROVED** [review]
LGTM. Just a minor nit.

File: ios/RNCAsyncStorage.m (hunk 1 above)
Comment (Krizzu) [reply ios/RNCAsyncStorage.m]: Good catch, fixed

**Pull request merged**